contracts/VoucherKernel.sol

```diff
@@ -624,15 +615,15 @@ contract VoucherKernel is IVoucherKernel, Ownable, Pausable, ReentrancyGuard, Us
             );            
         } else if (
             //if the opposite of what is the desired new state
-            isStatus(vouchersStatus[_tokenIdVoucher].status, (1-uint8(_newStatus)) + 2) // making it IDX_COMPLAIN or IDX_CANCEL_FAULT (opposite to new status) 
+            isStatus(vouchersStatus[_tokenIdVoucher].status, VoucherState((uint8(_newStatus) % 2 + 1))) // making it VoucherState.COMPLAIN or VoucherState.CANCEL_FAULT (opposite to new status) 
```

Comment:
It would be handy to add a comment about why  the `% 2 + 1` is needed.

Reply:
Done. Will be added in the next commit